demos/qml_optimization.ipynb

```diff
@@ -0,0 +1,469 @@
+        "    return loss"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "qUBCtuId62o9"
+      },
+      "source": [
+        "Note that the model above is just an example for demonstration -- there are important considerations that must be taken into account when performing QML research, including methods for data embedding, circuit architecture, and cost function, in order to build models that may have use. This is still an active area of research; see our [demonstrations](https://pennylane.ai/qml/demonstrations) for details.\n",
+        "\n",
+        "## Initialize your parameters\n",
+        "\n",
+        "Now, we can generate our trainable parameters `weights` and `bias` that will be used to train our QML model."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 7,
+      "metadata": {
+        "id": "sOWgLW5esIsO"
+      },
+      "outputs": [],
+      "source": [
+        "weights = jnp.ones([n_wires, 3])\n",
+        "bias = jnp.array(0.)\n",
+        "params = {\"weights\": weights, \"bias\": bias}"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "4mREqzeb7LI7"
+      },
+      "source": [
+        "Plugging the trainable parameters, data, and target labels into our cost function, we can see the current loss as well as the parameter gradients:"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 8,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "23t_bcyOtNBV",
+        "outputId": "7022b2d4-6c12-4a25-f903-00fd4e51be04"
+      },
+      "outputs": [
+        {
+          "data": {
+            "text/plain": [
+              "array(0.29232612)"
+            ]
+          },
+          "execution_count": 8,
+          "metadata": {},
+          "output_type": "execute_result"
+        }
+      ],
+      "source": [
+        "loss_fn(params, data, targets)"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 9,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "XCeL2_c4tPSi",
+        "outputId": "f405b448-494c-4e67-f187-e7a3adfdfe7b"
+      },
+      "outputs": [
+        {
+          "name": "stdout",
+          "output_type": "stream",
+          "text": [
+            "{'bias': array(-0.75432067), 'weights': array([[-1.95077271e-01,  5.28546590e-02, -4.89252073e-01],\n",
+            "       [-1.99687794e-02, -5.32871564e-02,  9.22904864e-02],\n",
+            "       [-2.71755507e-03, -9.64672786e-05, -4.79570827e-03],\n",
+            "       [-6.35443870e-02,  3.61110014e-02, -2.05196876e-01],\n",
+            "       [-9.02635405e-02,  1.63759364e-01, -5.64262612e-01]])}\n"
+          ]
+        }
+      ],
+      "source": [
+        "print(qml.qjit(catalyst.grad(loss_fn, method=\"fd\"))(params, data, targets))"
```

Comment:
```suggestion
        "print(qml.qjit(catalyst.grad(loss_fn))(params, data, targets))"
```
When https://github.com/PennyLaneAI/catalyst/pull/332 is merged :+1:

Reply:
@rmoyard I'll merge in this PR today; do you mind updating this demo in #332 before merging it in?